.gitattributes

```diff
@@ -0,0 +1 @@
+poetry.lock binary
```

Comment:
What exactly does this do?

Reply:
the description is in the PR description
```
I have told git (via the .gitattributes) file to treat the poetry.lock file as binary. This is a fairly standard practice, also followed in the JS community with the package-lock.json file. This does two primary things:
Tells GitHub to not attempt to display a diff for this file
Prevents frequent merge conflicts, since this tends to be a frequently updated file
```